src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -266,118 +258,29 @@ def create_empty_extra_products(self):
       classpath_product.update(compile_classpath)
 
   def select(self, target):
-    # Require that targets are marked for JVM compilation, to differentiate from
-    # targets owned by the scalajs contrib module.
-    if self._metacpable(target):
-      return True
     if not isinstance(target, JvmTarget):
       return False
-    return self._only_zinc_compilable(target) or self._rsc_compilable(target)
+    if self._classify_compile_target(target) is not None:
+      return True
+    return False
+
+  def _classify_compile_target(self, target):
```

Comment:
I might add a `TODO` for classifying targets based on tag.

Reply:
There's one on the enum call that this points to. I was thinking that I'd pass the tags etc there, but I could have that here instead.